tests/cxx/funcptrs-i1.h

```diff
@@ -44,11 +44,21 @@ Enum Function(Class*) {
   return E_one;
 }
 
+// Forward declaration means that FunctionReturningRecordType doesn't provide
+// its return type.
+class IndirectClass;
+IndirectClass FunctionReturningRecordType();
+
 template<typename T>
 int FunctionTemplate(Class*) {
   return T(10).Value();
 }
 
+template<typename T>
+T FunctionTemplate2() {}
+
+extern template IndirectClass FunctionTemplate2<IndirectClass>();
```

Comment:
Feels like it would be a stronger testcase if the template argument was different from the return type, e.g.
```
// here
template<typename T>
Retval FunctionTemplate2() {}

extern template Retval FunctionTemplate2<IndirectClass>();

// there
  // IWYU: FunctionTemplate2 is...*funcptrs-i1.h
  // IWYU: IndirectClass needs a declaration
  &FunctionTemplate2<IndirectClass>;
```

though I suspect that's not fully representative...? Might need something else than `Retval` that's forward-declarable for author intent, but even then I'm not sure if the template arg is somehow special?

Reply:
You're right, the template argument is unrelated to the problem. Besides, I've found out that there is one more case when return type full info is not needed: for references to non-explicit specializations when the primary template has no definition in the translation unit. I've added some code and a test case.